docs/serverless/03-02-function-specification.md

```diff
@@ -0,0 +1,136 @@
+    "ce-eventtypeversion": "v1",
+    "ce-specversion": "1.0",
+    "ce-id": "abc123",
+    "ce-time": "2020-12-20T13:37:33.647Z",
+    "data": {BUFFER},
+    "extensions": {
+        "request": {INCOMING_MESSAGE},
+        "response": {SERVER_RESPONSE},
+    }
+}
+```
+
+See the detailed descriptions of these fields:
+
+| Field | Description |
+|-------|-------------|
+| **ce-type** | Type of the CloudEvent data related to the originating occurrence |
+| **ce-source** | Unique context in which an event happened and can relate to an organization or a process |
+| **ce-eventtypeversion** | The version of the CloudEvent type |
+| **ce-specversion** | Version of the CloudEvent specification used for this event |
+| **ce-id** | Unique identifier of the event |
+| **ce-time** | Time at which the event was sent |
+| **data** | [Buffer](https://nodejs.org/api/buffer.html) that contains the request body. It can be either JSON or a string, depending on a request type |
+| **extensions** | JSON object that can contain event payload, a Function's incoming request, and an outgoing response |
+
+### Context object
+
+The `context` object contains information about the Function invocation, such as runtime details, execution timeout, or memory limits.
+
+See sample context details:
+
+```json
+...
+{ "function-name": "main",
+  "timeout": 180,
+  "runtime": "nodejs10",
+  "memory-limit": 200Mi }
+```
+
+See the detailed descriptions of these fields:
+
+| Field | Description |
+|-------|-------------|
+| **function-name** | Name of the invoked Function |
+| **timeout** | Time, in seconds, after which the system cancels the request to invoke the Function |
+| **runtime** | Environment used to run the Function. You can use `nodejs10`, `nodejs10`, or `python3.8`. |
+| **memory-limit** | Maximum amount of memory assigned to run a Function |  
+
+## Requests and responses
+
+Functions in Kyma support two execution types: **HTTP** (request or response) and **Events**. In both types, the **return** statement identifies a successful execution of the Function. For the event type, the event is reinjected until the execution is successful. Functions of the HTTP type can return data to the requesting entity.
+
+### HTTP requests
+
+You can use the **event.extensions.request** object to access headers, request a status, parse and access form data, cookies, file uploads, and other metadata. For more information, read the API documentation for [Node.js](https://nodejs.org/docs/latest-v12.x/api/http.html#http_class_http_clientrequest) and [Python](https://bottlepy.org/docs/dev/api.html#the-request-object).
+
+### Custom HTTP responses in Node.js
+
+By default, a failing Function simply throws an error to tell the event service to reinject the event at a later point. An HTTP-based Function returns the HTTP status code `500`. On the contrary, if you manage to invoke a Function successfully, the system returns the default HTTP status code `200`.
```

Comment:
```suggestion
By default, a failing Function simply throws an error to tell the event service to reinject the event at a later point. A failing HTTP-based Function returns the HTTP status code `500`. On the contrary, if you manage to invoke a Function successfully, the system returns the default HTTP status code `200`.
```

Reply:
maybe `such a` so as not to repeat `a failing`?